pennylane/logging/README.md

```diff
@@ -0,0 +1,307 @@
+```
+
+The first sections of the configuration file tell the logging infrastructure to avoid modification to existing log settings --- this is set to `true` by default for backwards compatibility, though can be problematic if using external packages. It is recommended to keep this as `false` unless required otherwise.
+
+
+```toml
+
+###############################################################################
+# Bind LogRecord filters defined in pennylane.logging module
+###############################################################################
+
+[filters]
+# Filter to show messages from the same local process as the Python script
+[filters.qml_LocalProcessFilter]
+"()" = "pennylane.logging.filter.LocalProcessFilter"
+
+# Filter to show debug level messages only
+[filters.qml_DebugOnlyFilter]
+"()" = "pennylane.logging.filter.DebugOnlyFilter"
+```
+The above section defines how to filter log messages (known as `LogRecords`), given some predicate. In this case, we have defined some classes, `LocalProcessFilter` and `DebugOnlyFilter` to filter based on process ID and on the severity of the incoming message. These can used in the next section.
+
+
+```toml
+###############################################################################
+# Bind handlers defined in the logging and in pennylane.logging modules
+###############################################################################
+
+[handlers]
+[handlers.qml_debug_stream]
+class = "logging.StreamHandler"
+formatter = "qml_default_formatter"
+level = "DEBUG"
+stream = "ext://sys.stdout"
+
+[handlers.qml_debug_stream_alt]
+class = "logging.StreamHandler"
+formatter = "qml_alt_formatter"
+level = "DEBUG"
+stream = "ext://sys.stdout"
+
+[handlers.qml_debug_file]
+class = "logging.handlers.RotatingFileHandler"
+formatter = "local_standard"
+level = "DEBUG"
+filename ='qml_debug.log' # use `/tmp/filename.log` on Linux machines to avoid long-term persistence
+maxBytes = 16777216 # 16MB per file before splitting
+backupCount = 10 # Create 'qml_debug.log.1', ... 'qml_debug.log.backupCount' files and rollover when maxBytes is reached
+
+[handlers.local_filtered_detailed_stdout]
+class = "logging.StreamHandler"
+formatter = "local_standard"
+level = "DEBUG"
+stream = "ext://sys.stdout"
+filters = ["qml_LocalProcessFilter", "qml_DebugOnlyFilter"]
+
+```
+The above defines how `LogRecord` messages are handled, and directs them to the appropriate sink. The logging framework supports many such directions (see [here](https://docs.python.org/3/library/logging.handlers.html) for more info), but for this example we have defined stream handlers (printing to the screen via the standard output), and a file handler with a size cap at 16MB. Each handler can be customized by filters and formatters so that the consumed message fits the needs of the user.
+
+```toml
+###############################################################################
+# Define logger controls for internal and external packages
```

Comment:
as a first-time user, I was expecting a global kind of "send my logs here" option, but that didn't seem to be the case. If I'm following, you define handlers above ("write to stdout", "write to qml_debug.log", etc.), then tell loggers which handlers to use, and that indirectly routes logs to output streams.

I guess what I'm wondering is:
- do we anticipate notable added value of splitting different log streams to different output streams?
- can we make it more obvious/clearly documented as to how you change where logs go? I'd personally want all logs dumped to a file so I can parse them later. if they're labelled properly, I wouldn't even care if they're all in the same place (`grep __main__ qml_debug.log > main_currtime.log`, etc.) but that might just be me.

Reply:
definitely. I'm happy to push this to another PR, but I have some ideas for what would be good. The README gives a good overview of fairly internal details, but I'd like to see something more friendly for someone who wants to use logging without understanding all the details. Should be hands-on examples like "write to stderr instead of stdout", "write all logs to a file", "split logs into different files for each interface", "change the verbosity everywhere/per-stream", etc. And we can stick that guide in the "Using PennyLane" category on our docs page.

Reply:
Yep, that's a good idea. I'll mark-out a docs page for this and we can identify what's a good level for "Getting started"